packages/core/src/validate.ts

```diff
@@ -111,10 +128,25 @@ export function getContractVersion(validation: Validation, contractName: string)
   return version;
 }
 
+function findContractByVersion(validation: Validation, version: Version): string | undefined {
+  return Object.keys(validation).find(name => validation[name].version?.withMetadata === version.withMetadata);
+}
+
+function isValidVersion(validation: Validation, version: Version): boolean {
+  return findContractByVersion(validation, version) !== undefined;
+}
+
+export function getValidVersion(validation: Validation, bytecode: string): Version {
+  let version: Version = getVersion(bytecode);
+  if (!isValidVersion(validation, version)) {
+    const unlinkedBytecode: string = getUnlinkedBytecode(validation, bytecode);
+    version = getVersion(unlinkedBytecode);
```

Comment:
In this case I would favor multiple return statements rather than the mutable `version` variable.

Reply:
Will take care of it. Do you think we should proceed with the refactor and handle this in `version.ts` ? It would be neater if you can simply query `getVersion` and get the correct one. (we also remove the `validVersion` naming which we know is not nice).